apps/rush-lib/src/cli/actions/PublishAction.ts

```diff
@@ -108,6 +115,22 @@ export default class PublishAction extends BaseRushAction {
       `the package is older than the current latest, so in publishing workflows for older releases, providing ` +
       `a tag is important. When hotfix changes are made, this parameter defaults to 'hotfix'.`
     });
+
+    // NPM pack tarball related parameters
+    this._pack = this.defineFlagParameter({
+      parameterLongName: '--pack',
+      description:
+        `Packs projects into tarballs instead of publishing to npm repository. It can only be used when ` +
+        `--include-all is specified. If this flag is specified, NPM registry related parameters will be ignored.`
+    });
+    this._releaseFolder = this.defineStringParameter({
+      parameterLongName: '--release-folder',
+      key: 'RELEASEFOLDER',
+      description:
+      `This parameter is used with --pack parameter to provide customized location for the tarballs instead of ` +
```

Comment:
>This parameter is used with --pack parameter  [](start = 7, length = 45)

If the person specifies "--release-folder" without "--pack", report an error like this:

"The --release-folder  option may only be used with the --pack flag."

Reply:
By default people should just use --pack and everything goes to the default common\temp folder.  --release-folder is only for customized cases.

---
In reply to: [176225647](https://github.com/Microsoft/web-build-tools/pull/579#discussion_r176225647) [](ancestors = 176225647)